File: go/gin/services/Crypt.go
package services

import "golang.org/x/crypto/bcrypt"

func GenerateAndValidate(plain []byte) func() {
	return func() {
		if hash, err := Encrypt(plain); err == nil {
			Validate(plain, hash)
		}
	}
}

func Encrypt(plain []byte) ([]byte, error) {
	return bcrypt.GenerateFromPassword(plain, 10)
}

func Validate(plain []byte, hash []byte) error {
	return bcrypt.CompareHashAndPassword(hash, plain)
}

File: stacks/rust/actix/src/crypt.rs
extern crate bcrypt;

use bcrypt::{hash, verify};

pub fn encrypt(plain: &str) -> String {
    return hash(plain, 10).unwrap();
}

pub fn validate(plain: &str, hashed: String) -> bool {
    return verify(plain, &hashed).unwrap();
}
